### PR TITLE
Git spec handling revamp

### PR DESCRIPTION
**Pull request by jim-minter**
xref https://github.com/openshift/origin/pull/15071

### CODE REVIEW

**Comment by jim-minter** [comment]
[test]

**Comment by jim-minter** [comment]
@bparees ptal...

**Comment by jim-minter** [comment]
@openshift/devex fyi

File: hack/test-stirunimage.sh [hunk 1]
```diff
@@ -82,11 +82,6 @@ test_debug "s2i build with relative path without file://"
 s2i build cakephp-ex openshift/php-55-centos7 test --loglevel=5 &> "${WORK_DIR}/s2i-rel-noproto.log"
 check_result $? "${WORK_DIR}/s2i-rel-noproto.log"
 
-test_debug "s2i build with relative path with file://"
-
-s2i build file://./cakephp-ex openshift/php-55-centos7 test --loglevel=5 &> "${WORK_DIR}/s2i-rel-proto.log"
-check_result $? "${WORK_DIR}/s2i-rel-proto.log"
```
Comment (jim-minter) [inline hack/test-stirunimage.sh]: yes, under the section marked "s2i build with relative path without file://" :-)

File: pkg/build/config.go [hunk 2]
```diff
@@ -41,13 +42,17 @@ func GenerateConfigFromLabels(config *api.Config, metadata *docker.PullResult) e
 	}
 
 	if repo, ok := labels[api.DefaultNamespace+"build.source-location"]; ok {
-		config.Source = repo
+		source, err := git.Parse(repo)
+		if err != nil {
+			return fmt.Errorf("couldn't parse label %q: %v", api.DefaultNamespace+"source-location", err)
```
Comment (jim-minter) [inline pkg/build/config.go]: fixed - thanks

**Comment by bparees** [comment]
[merge]

**Pull request merged**